lms/static/scripts/frontend_apps/components/LMSFilePicker.js

```diff
@@ -318,20 +319,33 @@ export default function LMSFilePicker({
   const { continueButton, warningOrError } =
     options[dialogState.continueAction];
 
-  if (dialogState.state === 'fetching') {
-    return null;
+  // During the first fetch and load, we don't know yet whether we'll have
+  // success fetching files, or whether we'll need auth'z or hit another
+  // error. To avoid hopscotching between modals, show only
+  // a full-screen spinner during this state.
+  const isLoading = dialogState.state === 'fetching';
+  // During subsequent file/folder fetches, represent the loading state within
+  // the file list UI (do not show a full-screen spinner)
+  const isReloading = dialogState.state === 'reloading';
+  // Render the file list and breadcrumbs UI in certain states. This also
+  // requires more real estate in the <Modal>, and applies an appropriate CSS
+  // class
+  const withFileUI = ['reloading', 'fetched'].includes(dialogState.state);
```

Comment:
Continuing to tack on state-related logic to this component is a bit inelegant, of course, but we know it's highly urgently due for a refactor, so I've left what I hope are thorough comments as to what is going on here. One of the underlying challenges is that we're asking a single component to be a whole lot of different things, and the Modal rendered below has to a lot of different things in a number of different states...